From 3d41efffffba010a00d646b8e03105e9701111e4 Mon Sep 17 00:00:00 2001
From: William Wu <william.wu@rock-chips.com>
Date: Thu, 29 Nov 2018 16:45:01 +0800
Subject: [PATCH] lib/Kconfig.debug: raise CONFIG_FRAME_WARN for Rockchip UVC

When building with the USB_CONFIGFS_F_UVC_ROCKCHIP, set the
default CONFIG_FRAME_WARN to 5120, since Rockchip UVC need
4096 bytes for UVC request data buffer, grow beyond 2048 bytes
stack size.

Change-Id: I2df8892b8e0c930053be0154609ccc1558318b33
Signed-off-by: William Wu <william.wu@rock-chips.com>
---
 lib/Kconfig.debug | 1 +
 1 file changed, 1 insertion(+)

diff --git a/lib/Kconfig.debug b/lib/Kconfig.debug
index 34c170a4a99d..c43d44e4efdf 100644
--- a/lib/Kconfig.debug
+++ b/lib/Kconfig.debug
@@ -204,6 +204,7 @@ config ENABLE_MUST_CHECK
 config FRAME_WARN
 	int "Warn for stack frames larger than (needs gcc 4.4)"
 	range 0 8192
+	default 5120 if USB_CONFIGFS_F_UVC_ROCKCHIP
 	default 2048 if GCC_PLUGIN_LATENT_ENTROPY
 	default 1024 if !64BIT
 	default 2048 if 64BIT
-- 
2.35.3

